projects/pgai/pgai/semantic_catalog/semantic_catalog.py

```diff
@@ -307,7 +307,24 @@ async def export_catalog(
         await async_export_to_yaml(yaml, load_from_catalog(catalog_con, self.id))
 
 
+async def _check_installed(con: CatalogConnection) -> None:
+    async with con.cursor() as cur:
+        await cur.execute(
+            """\
+            select count(*)
+            from information_schema.tables
+            where table_schema = %s and table_name = %s
+        """,
+            ("ai", "semantic_catalog"),
+        )
+        row = await cur.fetchone()
+        assert (
+            row is not None and row[0] > 0
+        ), "Semantic catalog is not installed, please run: pgai semantic-catalog create"
+
+
 async def from_id(con: CatalogConnection, id: int) -> SemanticCatalog:
+    await _check_installed(con)
```

Comment:
Instead of running an extra query to actively check, can we catch a "relation does not exist" exception specifically?

Reply:
Changed in 8bdde3e8d6d67492b2313a3e768bbfa5de0bbf96 to show the message off `psycopg.errors.UndefinedTable`, which should give the same behavior, though it's not quite as DRY.